Docs day acl improvements (#12460)
hashicorp/consul: website/content/docs/security/acl/acl-policies.mdx

```hcl
# These control access to the key/value store.
key_prefix "" {
  policy = "read"
}
key_prefix "foo/" {
  policy = "write"
}
key_prefix "foo/private/" {
  policy = "deny"
}
# Or for exact key matches
key "foo/bar/secret" {
  policy = "deny"
}

# This controls access to cluster-wide Consul operator information.
operator = "read"
```

</CodeBlockConfig>
<CodeBlockConfig lineNumbers>

```json
{
  "key": [
    {
      "foo/bar/secret": [
        {
          "policy": "deny"
        }
      ]
    }
  ],
  "key_prefix": [
    {
      "": [
        {
          "policy": "read"
        }
      ]
    },
    {
      "foo/": [
        {
          "policy": "write"
        }
      ]
    },
    {
      "foo/private/": [
        {
          "policy": "deny"
        }
      ]
    }
  ],
  "operator": "read"
}
```

</CodeBlockConfig>
</CodeTabs>

## Rule Scope

The rules from all policies, including roles and service identities, linked with a token are combined to form that token's effective rule set.
Policy rules can be defined in either an `allowlist` or `denylist` mode, depending on the configuration of the [`acl_default_policy`](/docs/agent/options#acl_default_policy).

> If the default policy is configured to deny access to all resources, then policy rules can be set to `allowlist` access to specific resources.
> If the default policy is to allow then access to all resources, policy rules can be used to explicitly deny access to resources.

Author: ```suggestion
If the default policy is configured to deny access to all resources, then you can specify `allowlist` in policy rules to explicitly allow access to resources.
Conversely, if the default policy is configured to allow access to all resources, then you can specify `denylist` in policy rules to explicitly deny access to resources.
```
